### PR TITLE
Add function to query keys

### PR DESCRIPTION
**Pull request by stoeffel**
This will be useful to clear out all `unsent-msg-*`

### CODE REVIEW

File: nri-redis/test/Main.hs [hunk 1]
```diff
@@ -261,7 +261,17 @@ queryTests redisHandler =
           |> Redis.query redisHandler
           |> Expect.succeeds
       field2
-        |> Expect.equal (Just "val2")
+        |> Expect.equal (Just "val2"),
+    Test.test "set and keys" <| \() -> do
+      Redis.set api "keys-1" "hello!" |> Redis.query testNS |> Expect.succeeds
+      Redis.set api "keys-2" "hello!" |> Redis.query testNS |> Expect.succeeds
```
Comment (juanedi) [inline nri-redis/test/Main.hs:267]: 🎨 adding a key that doesn't match would be nice to showcase the behavior of the function :-)

**Review by juanedi — COMMENTED** [review]
Code looks great! HOWEVER...

It looks like our current issues with HQE are because we are completely failing to write anything to Kafka. The way I see it we shouldn't need to make aide clear out unsent messages from redis after we fix that. Am I missing anything?

If that is correct, I think I would prefer not exposing `keys` (or at least not without any warnings) as folks may be tempted to use it incorrectly. Quoting [the redis docs](https://redis.io/commands/keys/):

> Warning: consider KEYS as a command that should only be used in production environments with extreme care. It may ruin performance when it is executed against large databases. This command is intended for debugging and special operations, such as changing your keyspace layout. Don't use KEYS in your regular application code. If you're looking for a way to find keys in a subset of your keyspace, consider using [SCAN](https://redis.io/commands/scan) or [sets](https://redis.io/topics/data-types#sets).

**Comment by stoeffel** [comment]
yes, I think you are right!

**Comment by stoeffel** [comment]
how annoying I can't close the PR on my phone

**Comment by juanedi** [comment]
haha on it 💛

**Pull request closed without merging**